assignments/asgmt6.rb

```diff
@@ -0,0 +1,86 @@
+ require 'minitest/autorun'
+  
+  class PriorityQueue
+
+    class SubQueue
+      attr :rank, :items
+      def initialize(rank)
+         @rank = rank
+         @items = []
+      end
+      def enqueue(item)
+        @items<<item
+      end
+      def dequeue
+        @items.shift
+      end
+      def empty?
+        @items.empty?
+      end
+      def peek
+        @items[0]
+      end
+      def length
+        @items.length
+      end
+    end #SubQueue
+    
+    def initialize
+      @high_sq = SubQueue.new(:high)
+      @medium_sq = SubQueue.new(:medium)
+      @low_sq = SubQueue.new(:low)
+      @sq_names = {high: @high_sq, medium: @medium_sq, low: @low_sq} 
+    end
+    def enqueue(item, priority=:medium)
+      subqueue = @sq_names.select {|i| i == priority}.values[0]
+      subqueue.enqueue(item)
+    end
+    def dequeue
+      if  @high_sq.length > 0
```

Comment:
instead of having to do this cascade of if statements, you could rely on the fact that dequeue returns null if the queue is empty. Then you could do the following:
`@high_sq.dequeue || @medium_sq.dequeue || @low_sq.dequeue`
Think about this. If you don't understand, ping me and I can explain.